### PR TITLE
[Translation zh-tw] updates > 2017 > 10 > remove-shadow-piercing.md 

### PR DESCRIPTION
**Pull request by limhenry**
[Check out a review request at GitLocalize](https://gitlocalize.com/repo/107/zh-TW/review/370)

### CODE REVIEW

**Comment by sotayamashita** [comment]
It fails because of `Error: OAuth2 authentication requires a token or key & secret to be set`

```
[10:06:37] Starting 'test:travis-init'...

[10:06:37] 'test:travis-init' errored after 28 ms

[10:06:37] Error: OAuth2 authentication requires a token or key & secret to be set

    at __dirname.authenticate (/home/travis/build/google/WebFundamentals/node_modules/github/lib/index.js:359:23)

    at Gulp.<anonymous> (/home/travis/build/google/WebFundamentals/gulp-tasks/test.js:1305:10)

    at module.exports (/home/travis/build/google/WebFundamentals/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:34:7)

    at Gulp.Orchestrator._runTask (/home/travis/build/google/WebFundamentals/node_modules/gulp/node_modules/orchestrator/index.js:273:3)

    at Gulp.Orchestrator._runStep (/home/travis/build/google/WebFundamentals/node_modules/gulp/node_modules/orchestrator/index.js:214:10)

    at Gulp.Orchestrator.start (/home/travis/build/google/WebFundamentals/node_modules/gulp/node_modules/orchestrator/index.js:134:8)

    at /home/travis/build/google/WebFundamentals/node_modules/gulp/bin/gulp.js:129:20

    at nextTickCallbackWith0Args (node.js:489:9)

    at process._tickCallback (node.js:418:13)

    at Function.Module.runMain (module.js:443:11)
```

File: src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{# wf_tags: webcomponents,shadowdom,style,css,deprecations,removals #}
+{# wf_featured_image: /web/updates/images/generic/styles.png #}
+{# wf_featured_snippet: 向shadow-piercing CSS選擇器說再見。 #}
+{# wf_blink_components: Blink>DOM #}
+
+# Chrome 63將刪除::shadow 和 /deep/ {: .page-title }
+
+{% include "web/_shared/contributors/robdodson.html" %}
+
+Translated by
```
Comment (petele) [inline src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md]: Per our [style guide](https://developers.google.com/web/resources/translations#credit), please put the translated by at the bottom of the article.

File: src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{% include "web/_shared/contributors/henrylim.html" %}
+
+在Chrome63之後，你將無法使用shadow-piercing 選擇器， `::shadow` 和 `/deep/` 來設計(style) shadow root裡面的內容。
+
+- `/deep/` 组合者(combinator)將會成為後代選擇器。 `x-foo /deep/ div` 將會當成 `x-foo div`.
+- `::shadow` 偽元素(pseudo-element)將不會匹配任何元素。
+
+注：如果你的網頁使用Polymer, Polymer團隊已發布了一個[如何讓你遷移](https://www.polymer-project.org/blog/2017-10-18-upcoming-changes.html) `::shadow` 和 `/deep/`的文章。
```
Comment (petele) [inline src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md]: [Callouts](https://developers.google.com/web/resources/markdown-syntax#callouts) like `Note:` should NOT be translated, they're handled differently by the markdown processor.

Comment (petele) [inline src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md]: Please break lines at [80 chars](https://developers.google.com/style/html-formatting#line-length)

File: src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## 決定刪除::shadow 和 /deep/ 的原因
+
+`::shadow` 和 `/deep/` 已經在Chrome 45被棄用了。這是在[2015年4月由Web組件聚會](https://www.w3.org/wiki/Webapps/WebComponentsApril2015Meeting)中的全部的出席者決定的。
+
+目前，shadow-piercing選擇器的問題是它違反了封裝(encapsulation)以及在某些時候組件無法更改組件的內部實施。
+
+注：目前JavaScript API如`querySelector()` 和 `querySelectorAll()`會繼續支持`::shadow` 和 `/deep/`。目前這API的支持[正在在GitHub上討論](https://github.com/w3c/webcomponents/issues/78)。
```
Comment (petele) [inline src/content/zh-tw/updates/2017/10/remove-shadow-piercing.md]: [Callouts](https://developers.google.com/web/resources/markdown-syntax#callouts) like `Note:` should NOT be translated, they're handled differently by the markdown processor.

**Pull request closed without merging**